### ISSUE TITLE
Incorrect code completion when backticks are present in data.frame column name

### ISSUE THREAD

**Issue by ToeKneeFan** [rstudio/rstudio#4094]
Likely a very rare use case (or maybe an intended behavior to dissuade ill-advised naming practices), but RStudio does not appear to correctly recognize that literal backtick characters can be present in data.frame column names.

### System details

    RStudio Edition : Desktop
    RStudio Version : RStudio Preview Version 1.2.1114
    OS Version      : Mac OS 10.14.2
    R Version       : 3.5.1

### Steps to reproduce the problem

1. Generate a data.frame containing a column named with a literal backtick character.
2. Type the name of the data.frame and `$` to trigger code completion.
```
df <- data.frame(`\`Col with backtick\`` = 1:3, check.names = F)
df$
```

### Describe the problem in detail

RStudio suggests a column name that is not present in the data.frame due to a failure to recognize the backtick characters. For example, the above code results in the suggestion `` df$`Col with backtick` ``, which has the value `NULL`, as it is not present in the data.frame. 

### Describe the behavior you expected

We would not expect RStudio to suggest a column name that does not exist. Each of the following behaviors would be less astonishing

1. Suggest the actual column name, escaping the backtick characters as necessary.
2. Give no code-completion suggestion for columns containing a backtick character.
3. Give the original code-completion suggestion but warn the user that the suggested column name will be incorrect due to a syntactically invalid or deprecated column name.

Options 2 and 3 are less convenient but would at least prevent or warn against erroneous code completion.

**Comment by ToeKneeFan** [comment]
This appears to be an ongoing issue in RStudio v1.4.1103. The code-completion suggestion is still "fooled" by the presence of backticks in a column name, giving a suggestion for a nonexistent column. 

It now yields a suggestion for ` df$``Col with backtick`` `, which yields an error `Error: attempt to use zero-length variable name` upon printing.

**Comment by ronblum** [comment]
Issue #11689 is a duplicate of this issue, but that one has a potential fix associated with it so I'm closing this one.

**Comment by ToeKneeFan** [comment]
This one has a potential fix as well (though perhaps less convenient to implement). If backticks within names are automatically escaped, then the suggestion would work fine. e.g., if typing `df$` caused the suggestion
```
      df$`\`Col with backtick\``
```
then the suggestion would work.

**Comment by ronblum** [comment]
My mistake: The problems aren't duplicates, as noted[ in the comment of issue #11689](https://github.com/rstudio/rstudio/issues/11689#issuecomment-1203309396). Reopening.

**Comment by romainfrancois** [comment]
It seems fixed in : RStudio 2023.05.0-daily+336 "Mountain Hydrangea" Daily (afdee25b92abb127646fc345f5be6481ea777a0f, 2023-05-11) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) RStudio/2023.05.0-daily+336 Chrome/110.0.5481.208 Electron/23.3.0 Safari/537.36

<img width="746" alt="image" src="https://github.com/rstudio/rstudio/assets/2625526/9cc4e3b3-42f8-44f2-be6a-a80830d63f57">

**Comment by ToeKneeFan** [comment]
You're right! It seems that suggestions containing backticks are automatically escaped now.
